### DISCUSSION TITLE
Ethemeral host?

### DISCUSSION THREAD

**Discussion by brigidspeed23** [basecamp/kamal#605] [General]
We're in the process of implementing Kamal with our Rails app running in AWS. We are particularly interested in the scenario where we lose our EC2 host unexpectedly (eg hardware failure in AWS) and how to implement an autorecovery of the host with minimal downtime (we have the Kamal host running in an Autoscaling group currently so the EC2 replacement itself is handle seamlessly). Our understanding is that the host is stateful (files in the .kamal directory). If anyone else has tackled this already and has any hints or suggestions it would be much appreciated.

**Comment by willzoltan** [comment]
This is a great question, and I don't have any answers, but it is definitely a concern of ours about moving to Kamal. @brigidspeed23 did you find a solution?

**Reply by brigidspeed23** [reply]
No will didn't find a solution. We've backed away from implementing Kamal in a HA prod environment, although we do still run it in some of our less critical lower environments. I'm sure there is a solution for storing the Kamal state off the instance and implementing auto-recovery logic to install the state back onto a replacement instance, but we thought it was probably pushing Kamal into use cases it wasn't designed for.
